.github/workflows/run-benchmark.yml

```diff
@@ -21,48 +31,102 @@ jobs:
       - name: Configure RunsOn
         uses: runs-on/action@v1
 
+      # jemalloc is a dependency of the Benchmarking package
+      # actions/cache will detect zstd and will become much faster.
+      - name: Install jemalloc and zstd
+        run: |
+          apt update -q && apt install -yq libjemalloc-dev zstd curl jq
+
+      - name: Find commit info
+        id: commit-info
+        run: |
+          if [[ -n '${{ inputs.sha }}' ]]; then
+            FULL_SHA="$(curl "https://api.github.com/repos/vapor/multipart-kit/commits/${{ inputs.sha }}" | jq -r '.sha')"
+
+            # Get the oldest PR that contains the commit
+            PR="$(
+              curl "https://api.github.com/search/issues?q=${FULL_SHA}" | \
+                jq 'select(.items[].pull_request.url | contains("/${{ github.repository }}/")) | .items[-1]'
+            )"
+
+            PR_NUMBER="$(echo "${PR}" | jq -r '.number')"
+            PR_TITLE="$(echo "${PR}" | jq -r '.title')"
+
+            echo "ref=${FULL_SHA}" >> "${GITHUB_OUTPUT}"
+            echo "sha=${FULL_SHA}" >> "${GITHUB_OUTPUT}"
+            echo "pr_number=${PR_NUMBER}" >> "${GITHUB_OUTPUT}"
+            echo "pr_title=${PR_TITLE}" >> "${GITHUB_OUTPUT}"
```

Comment:
`ref` to pass to actions/checkout.
`sha` for cache/restore.
`pr_number` to post the comment in.
`pr_title` as the baseline name.